### DISCUSSION TITLE
WebView Renderer Migration does not load HTML or URL content on Maui compatibility renderer

### DISCUSSION THREAD

**Discussion by DanielCauser** [dotnet/maui#18808] [Upgrading from Xamarin]
Hello!

I'm currently working on migrating our Xamarin Forms app into .net Maui. That said, I'm migrating the Xamarin Forms renderers and using the Compatibility path to simplify the migration process.

I created this discussion because I'm not sure if it is a bug or I'm missing something on my migration.

 Here is the outcome of the migration of the Xamarin Forms into .Net MAUI. As you can see, the renderer in the XF platform loads the HTML content, whereas rendererer in the .NET MAUI compatibility mode does not.
 
![Screenshot 2023-11-16 at 10 16 45 AM](https://github.com/dotnet/maui/assets/7211075/c98d4906-b8d6-4549-a509-f9ad5de1a217)

I'm also attaching the zip of the projects I'm using to replicate the issue.
[CustomRenderer.zip](https://github.com/dotnet/maui/files/13379519/CustomRenderer.zip)
The renderer looks like the one bellow. I pick the properties from the shared control and render them on the native webview.
```
namespace XamarinCustomRenderer.iOS.Renderers
{
    public class HybridWebViewRenderer : ViewRenderer<HybridWebView, WKWebView>
    {

        protected override void OnElementChanged(ElementChangedEventArgs<HybridWebView> e)
        {
            base.OnElementChanged(e);

            if (Control == null && e.NewElement != null)
            {
                _userController = new WKUserContentController();

                var config = new WKWebViewConfiguration
                {
                    UserContentController = _userController,
                    IgnoresViewportScaleLimits = true
                };

                var webView = new WKWebView(Frame, config)
                {
                    NavigationDelegate = new HybridWkWebViewDelegate(this),
                    UIDelegate = new WKWebViewUIDelegate()
                };

                webView.ScrollView.ShowsVerticalScrollIndicator = false;
                webView.ScrollView.ShowsHorizontalScrollIndicator = false;

                webView.LoadHtmlString(new NSString(e.NewElement.HtmlContent), null);
                //var link = new NSUrl(e.NewElement.URL);
                //var request = new NSUrlRequest(url: link);
                //webView.LoadRequest(request);

                SetNativeControl(webView);
            }
        }  
```

**Comment by DanielCauser** [comment]
Maybe the `ViewRenderer ` fallls under this rule?
![Screenshot 2023-11-16 at 10 59 46 AM](https://github.com/dotnet/maui/assets/7211075/b55fcedd-e428-41a7-bac4-1c6dde3f5497)

**Reply by DanielCauser** [reply]
Maybe not since `ViewRenderer` inherits directly from `VisualElementRenderer`
https://github.com/dotnet/maui/blob/573dfb92f2ec621b6d8f5c4d36b45e8c47da1a85/src/Controls/src/Core/Compatibility/Handlers/iOS/ViewRenderer.cs#L14

**Comment by sawankumarbundelkhandi** [comment]
@DanielCauser did you find any solution?

**Reply by DanielCauser** [reply]
I've found a work around. Here is the answer!

https://github.com/dotnet/maui/issues/18842

**Reply by sawankumarbundelkhandi** [reply]
Thank you @DanielCauser. It's working now.
